### PR TITLE
Remove generated `neil` code from Git repo

### LINKED ISSUE #70: Remove generated `neil` code from Git repo
See here: https://github.com/babashka/neil/pull/67#issuecomment-1203663690

### PR DESCRIPTION
**Pull request by rads**
Resolves #70.

### CODE REVIEW

File: neil [hunk 1]
```diff
@@ -1,852 +1,14 @@
 #!/usr/bin/env bb
 ;; -*- mode: clojure -*-
 
-;; DO NOT EDIT DIRECTLY. GENERATED FROM scr/babashka/neil.clj
+(def deps
+ '{org.babashka/neil {:git/url "https://github.com/babashka/neil.git"
+                      :git/tag "v0.1.41" :git/sha "77288d4"}
+   org.babashka/spec.alpha {:git/url "https://github.com/babashka/spec.alpha"
+                            :git/sha "1a841c4cc1d4f6dab7505a98ed2d532dd9d56b78"}})
```
Comment (rads) [inline neil:8]: I had to put the `org.babashka/spec.alpha` dependency here or else the `clj -M:neil` test would fail.

File: neil-local [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bb
+;; -*- mode: clojure -*-
+
+(require '[babashka.fs :as fs])
+
+(load-file (str (fs/parent *file*) "/neil"))
+(deps/merge-deps '{:deps {org.babashka/neil {:local/root "."}}})
```
Comment (borkdude) [inline neil-local:7]: merge deps is just a pure function, it doesn't do any side effects

**Comment by borkdude** [comment]
@rads We would still need to have the `neil` script but with `babashka.deps/add-deps` of the neil repo itself right?

**Comment by rads** [comment]
@borkdude:
> @rads We would still need to have the neil script but with babashka.deps/add-deps of the neil repo itself right?

Yeah, that's how it works in this PR. It may be hard to see because GitHub is hiding the diff since it's so large: https://github.com/babashka/neil/blob/8fa83b522dbdecb330fe0e5755a58dbcc32dea9c/neil

Working on fixing merge conflicts now.

**Comment by rads** [comment]
To be clear, the `neil` script still exists, but the `neil-local` script was added to wrap it so we can use `:local/root` for running tests and development.

**Comment by borkdude** [comment]
Oh right!

**Comment by rads** [comment]
One caveat I just thought of: we won't be able to continue using `:git/tag` inside the `neil` script because it would require us to know the SHA _before_ the tag is created for a new release since `:git/sha` is also required. We cannot create the tag until after the `neil` change happens, but the `neil` change can't happen until the tag is created.

I think the solution here is to use the full `:git/sha` instead of `:git/tag` inside `neil`.

**Comment by rads** [comment]
Another way to work around this would be to create a "pre-release" tag that `neil` would be able to reference before creating the "release" tag for package managers. Not sure if this is better than just using the full SHA though.

**Comment by borkdude** [comment]
Hmm... tricky. Maybe uploading the script or using the uberjar approach (like done with http-server [here](https://github.com/babashka/http-server/releases) is a better idea then - then we'd just make one breaking change. I control the brew package so this isn't a problem, but scoop and nix have to be updated.

**Comment by rads** [comment]
Going to continue with the uberjar approach in this PR: https://github.com/babashka/neil/pull/75

**Comment by rads** [comment]
Oops, I meant this PR: https://github.com/babashka/neil/pull/78 (fixed my earlier comment as well)

**Pull request closed without merging**